hydra-redis/redisgraph_demo1.py

```diff
@@ -0,0 +1,294 @@
+    """
+    print("accessing every collection in entrypoint")
+    for obj in collection_endpoint:
+        obj_method = []
+        properties = []
+        obj_properties = {}
+        print(
+            "check url for endpoint",
+            base_url +
+            collection_endpoint[obj])
+        new_url = base_url + collection_endpoint[obj]
+        for obj2 in api_doc.collections[obj]["collection"].supportedOperation:
+            obj_method.append(obj2.method)
+        obj_properties["methods"] = str(obj_method)
+        print(obj_properties["methods"])
+        for obj2 in api_doc.collections[obj]["collection"].supportedProperty:
+            properties.append(str(obj2.desc[4:]))
+        obj_properties["title"] = str(properties)
+        response = ur.urlopen(base_url + collection_endpoint[obj])
+        new_file = json.loads(response.read().decode('utf-8'))
+        obj_properties["members"] = str(new_file["members"])
+        obj_collection_node = Node(
+            label="id", alias=obj, properties=obj_properties)
+        redis_graph.add_node(obj_collection_node)
+        edge = Edge(url_node, "has_collection", obj_collection_node)
+        redis_graph.add_edge(edge)
+        collectionobjects(
+            obj_collection_node,
+            new_file["members"],
+            new_url,
+            url_node)
+
+
+def get_apistructure(entrypoint_obj, url_node):
+    """ It breaks the endpoint into two parts collection and classes"""
+    collection_endpoint = {}
+    classes_endpoint = {}
+    collection = 0
+    classes = 0
+    print("split entrypoint into 2 types of endpoints collection and classes")
+    for obj in url_node.properties:
+        if obj != "@id" and obj != "@type" and obj != "@context":
+            if obj in api_doc.collections.keys():
+                collection = 1
+                collection_endpoint[obj] = str(url_node.properties[obj])
+            else:
+                classes = 1
+                classes_endpoint[obj] = str(url_node.properties[obj])
+    if collection == 1:
+        endpointCollection(collection_endpoint, url_node)
+
+    if classes == 1:
+        endpointclasses(classes_endpoint, url_node)
+
+
+def get_endpoints(entrypoint_obj):
+    """create node for entrypoint"""
+    print("creating entrypoint node")
+    url_node = Node(label="id", alias="Entrypoint", properties=entrypoint_obj)
+    redis_graph.add_node(url_node)
+    return get_apistructure(entrypoint_obj, url_node)
+
+
+if __name__ == "__main__":
+    url = "http://35.224.198.158:8080/api"
+    match_obj = re.match(r'(.*)://(.*)/(.*)/?', url, re.M | re.I)
+    base_url = "{0}://{1}".format(match_obj.group(1), match_obj.group(2))
+    entrypoint = "/" + match_obj.group(3) + "/"
+    print("baseurl, entrypoint ", base_url, entrypoint)
+
+    apidoc = final_file(url + "/vocab")
+    api_doc = doc_maker.create_doc(apidoc)
+    get_endpoints(api_doc.entrypoint.get())
+    print("commiting")
+    redis_graph.commit()
```

Comment:
Again, we have no way of verifying what the graph is and how it is being stored, this was one of the tasks you were supposed to complete this week.

Reply:
In line 293, there are two commenting lines actually if you run the file with uncommenting these lines it will show the all the edges between all the nodes in the graph. So, it can be a look at the graph how it's look like and for getting the data in the graph I'll update it early.

Reply:
There are a lot of objects being added, possibly reduce them? We just need to make sure it works for a few different types, no need to add 30-40 objects. I suggest you try using **graphviz** for the visualisation, it is extremely simple, just a few lines of code need to be added and you can create a `.png` file showing the graph.  [example](https://github.com/pygraphviz/pygraphviz/blob/master/examples/simple.py)

Reply:
Here are some graph structures which is generated from graphviz with the code. https://docs.google.com/document/d/1h1EER1I2QXE7P2WorX7DTuUuLNIXIEqcX1ptoKXp5io/edit?usp=sharing
I have also added some lines in the code for getting the whole graph structure using graphviz.